woocommerce/woocommerce
# Refactoring LayoutContext and implementing with product editor

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

As per the issue, we occasionally have the need to know "where we are" in the application, when using common components across different sections. Without complex prop drilling we can use the `LayoutContext` component. In this PR I've moved this to the `admin-layout` package, refactored the component, updated references, and implemented on the blocks editor.

Closes #37740 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout branch and enable the `new-product-management-experience` feature flag.
2. Run `localStorage.setItem( 'debug', 'wc-admin:tracks' );` in the console.
3. Navigate to WooCommerce -> Home
4. Ensure the `wcadmin_tasklist_view` event fires with the property `context: woocommerce-home`
5. Click on "Add Product" item in task list, and ensure that `wcadmin_tasklist_click` event fires with the same `context` property.
6. Navigate to the Customers screen and click the "Finish Setup" button in the Activity Panel to display the task list in the side bar.
7. Ensure the `wcadmin_tasklist_view` event fires, but this time the `context` property should be `woocommerce-analytics-customers/activity-panel`
8. Click the "Add a product" task again and ensure the `wcadmin_tasklist_click` has the same context property.
9. Navigate to Products -> Add New, and you should see the old React product editor.
10. Click the "More" menu within the Header and click "Share feedback"
11. Enter any information and click "Share"
12. The `wcadmin_ces_feedback` even should fire with the `block_editor: false` property.
13. Enable the `block-editor-feature-enabled` feature flag, and refresh the product editor to see the new product block editor.
14. Hit the more menu again, and complete the "Share feedback" modal again.
15. After clicking "Share" you should see the same `wcadmin_ces_feedback` event, but with `block_editor` equal to `true`.
